nl_server/embeddings_store.py

```diff
@@ -49,6 +56,24 @@ def __init__(self, indexes: List[EmbeddingsIndex]):
   def get(self, index_type: str = DEFAULT_INDEX_TYPE) -> Embeddings:
     return self.embeddings_map.get(index_type)
 
+  def merge_custom_index(self, custom_idx: EmbeddingsIndex):
+    """Merges the specified custom index with the default index.
+
+    This method will be called if a new custom index is loaded at runtime
+    via the /api/load/ call.
+    """
+    default_idx = dataclasses.replace(self.original_default_idx)
```

Comment:
Is the intent of `dataclasses.replace` the same as doing `copy.copy` / `copy.deepcopy`?  Maybe prefer the latter?

Reply:
Done.